### PR TITLE
Branch date

### PR DESCRIPTION
**Pull request by mcmc101001**

### CODE REVIEW

**Review by Xuan127 — APPROVED** [review]
LGTM

File: src/main/java/cashleh/parser/Parser.java [hunk 1]
```diff
@@ -52,63 +54,64 @@ public Command parse(String input) throws CashLehException {
...
 
 
     private Expense getExpense(String input) throws CashLehParsingException {
-        if (!input.contains("/amt")) {
-            throw new CashLehParsingException("Please enter a proper amount using '/amt'.");
-        }
-        String[] expenseInfo = input.substring(ADD_EXPENSE.length() + 1).split("/amt", 2);
-        String expenseName = expenseInfo[0].trim();
-        String expenseAmtString = expenseInfo[1].trim();
+        String[] format = {ADD_EXPENSE, "/amt", "/date:optional"};
```
Comment (danielpappa) [inline src/main/java/cashleh/parser/Parser.java:63]: Very nice way of handling input data!

File: src/main/java/cashleh/parser/Parser.java [hunk 2]
```diff
@@ -52,63 +54,64 @@ public Command parse(String input) throws CashLehException {
...
     }
 
     private Income getIncome(String input) throws CashLehParsingException {
-        if (!input.contains("/amt")) {
-            throw new CashLehParsingException("Please enter a proper amount using '/amt'.");
-        }
-        String[] incomeInfo = input.substring(ADD_INCOME.length() + 1).split("/amt", 2);
-        String incomeName = incomeInfo[0].trim();
-        String incomeAmtString = incomeInfo[1].trim();
+        String[] format = {ADD_INCOME, "/amt", "/date:optional"};
```
Comment (danielpappa) [inline src/main/java/cashleh/parser/Parser.java:90]: Very nice way of parsing input data!

File: src/main/java/cashleh/parser/StringTokenizer.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    private static void checkAllPrefixesPresent(String input, String[] prefixes) throws CashLehParsingException {
+        for (int i = 0; i < prefixes.length; i++) {
+            if (prefixes[i].contains(":optional")) {
+                prefixes[i] = prefixes[i].replace(":optional", "");
+                return;
+            }
+            if (!input.contains(prefixes[i])) {
+                throw new CashLehParsingException("Aiyoh! Your input blur like sotong... " +
```
Comment (danielpappa) [inline src/main/java/cashleh/parser/StringTokenizer.java:36]: Should maybe avoid using singlish in output...

File: src/main/java/cashleh/transaction/Transaction.java [hunk 4]
```diff
@@ -1,13 +1,35 @@
...
+
+    /*
+     * Constructor for Transaction - set date to current date if not specified
+     * @param description Description of the transaction
+     * @param amount Amount of the transaction
+     */
     public Transaction(String description, double amount) {
         this.description = description;
         this.amount = amount;
+        this.date = LocalDate.now();
```
Comment (danielpappa) [inline src/main/java/cashleh/transaction/Transaction.java:20]: Great idea!

**Review by danielpappa — APPROVED** [review]

**Pull request merged**